pkg/provisioner/controller.go

```diff
@@ -331,18 +334,24 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 		if err != nil {
 			return fmt.Errorf("error updating OB %q's status to %q: %v", ob.Name, v1alpha1.ObjectBucketStatusPhaseBound, err)
 		}
+		ob = createdOB // don't let ob become nil but allow it to be updated on success
+
 		// update OBC
+		// Note: just as with the ob, don't all obc to be nil if returning for error for cleanup
+		// because obc has important information also
```

Comment:
how does the obc become nil?

Reply:
If updating the obc fails, `obc` becomes `nil` because the fxn returns `nil, <some error>`